op-node/rollup/sequencing/origin_selector_test.go

```diff
@@ -189,6 +196,28 @@ func TestOriginSelectorAdvances(t *testing.T) {
 	next, err = s.FindL1Origin(ctx, l2Head)
 	require.Nil(t, err)
 	require.Equal(t, c, next)
+
+	// Now force the retrieval of the next L1 origin
+	l1.ExpectL1BlockRefByHash(c.Hash, c, nil)
+	// It may not be ready yet. Expect an error, even if we could stick to c.
+	l1.ExpectL1BlockRefByNumber(d.Number, eth.BlockRef{}, ethereum.NotFound)
+
+	// at origin d, and can stick to c (thanks to timestamp), if it wasn't for recover mode.
+	s.recoverMode = true
+	l2Head = eth.L2BlockRef{
+		L1Origin: c.ID(),
+		Time:     d.Time + 4,
+	}
+	_, err = s.FindL1Origin(ctx, l2Head)
+	require.ErrorIs(t, err, derive.ErrTemporary)
+	require.ErrorIs(t, err, ethereum.NotFound)
+
+	// Now actually get to L1 origin d.
+	l1.ExpectL1BlockRefByHash(c.Hash, c, nil)
+	l1.ExpectL1BlockRefByNumber(d.Number, d, nil)
+	next, err = s.FindL1Origin(ctx, l2Head)
```

Comment:
So without recovery mode enabled, this would return `c` still? Could we add this as a pre check before enabling the recovery mode, so that it's clear from the test what the recover mode changes in behavior?
Maybe add a bit more doc on why this tests the recovery mode properly.

Reply:
Made this clearer here [9ae3773](https://github.com/ethereum-optimism/optimism/pull/14823/commits/9ae377348481907d5b469a08c4b658e8a14099c7) (there is now a subtest for recover mode ON and OFF with different assertions and mocks).